worker/uniter/runner/context/relation.go

```diff
@@ -15,14 +15,66 @@ import (
 	"github.com/juju/juju/worker/uniter/runner/jujuc"
 )
 
+type Relation interface {
```

Comment:
Is there a reason we don't have these interfaces in `interface.go`?

Reply:
interface.go for this pr is in a different package, relation.   I wasn't focused on interfaces for the context package.  They can be moved together in a next step.